website/src/docs/hotchocolate/api-reference/migrate-from-11-to-12.md

```diff
@@ -45,3 +45,110 @@ services
     .AddGraphQLServer()
     .AddType(() => new UrlType("Url"));
 ```
+
+# Relay
+
+## Configuration
+
+Previously the configuration of the Relay integration was focused around the `EnableRelaySupport()` method. It allowed you to enable Global Object Identification and automatically adding a query field to mutation payloads.
+
+The problem is that `EnableRelaySupport()` always enables the Global Object Identification feature. This is not obviously implied by the name and also prevents you from using the other feature in isolation.
+
+Therefore we introduced two separate APIs to give you more explicit control over which parts of the Relay integration you want to enable.
+
+### Global Object Identification
+
+**OLD**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .EnableRelaySupport();
+```
+
+**NEW**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .AddGlobalObjectIdentification();
+```
+
+### Query field in Mutation payloads
+
+**OLD**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .EnableRelaySupport(new RelayOptions
+    {
+        AddQueryFieldToMutationPayloads = true,
+        QueryFieldName = "rootQuery",
+        MutationPayloadPredicate = type => type.Name.Value.EndsWith("Result")
+    });
+```
+
+**NEW**
+
+```csharp
+sevices
+    .AddGraphQL()
+    .AddQueryFieldToMutationPayloads(options =>
+    {
+        options.QueryFieldName = "rootQuery";
+        options.MutationPayloadPredicate =
+            type => type.Name.Value.EndsWith("Result");
+    });
+```
+
+If you just want to enable the feature without further configuration, you can omit the `options =>` action.
+
+> ⚠️ Note: Since `EnableRelaySupport()` previously always implied the usage of Global Object Identification, you might have to enable Global Object Identification separately as well.
+
+## ID
+
+We renamed the `ID` attribute and extension methods to `GlobalId`, in order to more accurately reflect their connection to the Global Object Identification feature.
+
+**OLD**
+
+```csharp
+[ID]
+public string Id { get; set; }
+```
+
+```csharp
+descriptor.Field(f => f.Id).ID();
+```
+
+**NEW**
+
+```csharp
+[GlobalId]
+public string Id { get; set; }
+```
+
+```csharp
+descriptor.Field(f => f.Id).GlobalId();
+```
+
+### Shorthand for specifying the IdType
+
+If you have previously used the `IDAttribute` as a shorthand for the `ID` scalar (`IdType`), but didn't actually need the Relay specific encoding of the Ids, you can create an own attribute and use it as shorthand:
+
+```csharp
+public class IdAttribute : ObjectFieldDescriptorAttribute
+{
+    public override void OnConfigure(IDescriptorContext context,
+        IObjectFieldDescriptor descriptor, MemberInfo member)
+    {
+        descriptor.Type<NonNullType<IdType>>();
```

Comment:
Hmm, maybe this needs to be more complicated. If you do .ID() right now, without nullable enabled, does it always type to `ID`? Forcing you to then make it non-null with another extension method call? 

Potentially this needs to be `.Type<IdType>>()` and then that'd use nullable reference types where possible and then it'd act like before.

Not too sure.

Reply:
Yes, you're right type modifiers are currently respected by the ID attribute:https://github.com/ChilliCream/hotchocolate/blob/d84041d7059e77b5af2edd8c17df0fe98d2adf35/src/HotChocolate/Core/src/Types/Types/Relay/Extensions/RelayIdFieldExtensions.cs#L85
I thought it doesn't make sense for an ID to be nullable, but I forgot that these methods are also used for arguments.
Including all of this rewrite code in the migration guide would be a bit much though. Should we maybe just link the code to the IDAttribute and these extension methods? 
I also think we can cover type rewriting in the documentation at some point.

Reply:
Also they are used for foreign keys.